packages/protocol-perf/src/index.ts

```diff
@@ -126,19 +133,54 @@ class DefaultPerfService implements Startable, PerfService {
       throw new Error('bytesToSendBack was not set')
     }
 
+    let lastAmountOfBytesSent = BigInt(0)
+    let lastReportedTime = Date.now()
+    let totalBytesSent = BigInt(0)
+
+    const initialStartTime = Date.now()
+
     await stream.sink(async function * () {
       while (bytesToSendBack > 0n) {
         let toSend: bigint = writeBlockSize
         if (toSend > bytesToSendBack) {
           toSend = bytesToSendBack
         }
+
         bytesToSendBack = bytesToSendBack - toSend
         yield uint8Buf.subarray(0, Number(toSend))
+
+        if (Date.now() - lastReportedTime > 1000) {
+          const output: PerfOutput = {
+            type: 'intermediary',
+            timeSeconds: (Date.now() - lastReportedTime) / 1000,
+            uploadBytes: BigInt(0).toString(),
+            downloadBytes: lastAmountOfBytesSent.toString()
+          }
+
+          // eslint-disable-next-line no-console
+          console.log(JSON.stringify(output))
+
+          lastReportedTime = Date.now()
+          lastAmountOfBytesSent = BigInt(0)
+        }
```

Comment:
```suggestion
        if (Date.now() - lastReportedTime > 1000) {
          const newReportedTime = Date.now()
          const output: PerfOutput = {
            type: 'intermediary',
            timeSeconds: (newReportedTime - lastReportedTime) / 1000,
            uploadBytes: BigInt(0).toString(),
            downloadBytes: lastAmountOfBytesSent.toString()
          }

          // eslint-disable-next-line no-console
          console.log(JSON.stringify(output))

          lastReportedTime = newReportedTime
          lastAmountOfBytesSent = BigInt(0)
        }
```

move up lastReportedTime date so it doesn't have microsecond lag from console.log+stringify

Reply:
idk why the diff here is showing so many changes. Only changed line 153, 156, & 164